src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -32,14 +32,63 @@
 
 namespace Microsoft.ML.Transforms.Conversions
 {
+    /// <summary>
+    /// The ValueMappingEstimator is a 1-1 mapping from a key to value. This particular class load the mappings from an <see cref="IDataView"/>.
+    /// This gives user the flexibility to load the mapping from file instead of using IEnumerable in <see cref="ValueMappingEstimator{TKey, TValue}"/>
+    /// </summary>
+    public class ValueMappingEstimator : TrivialEstimator<ValueMappingTransformer>
+    {
+        private readonly (string input, string output)[] _columns;
+
+        /// <summary>
+        /// Constructs the ValueMappingEstimator, key type -> value type mapping
+        /// </summary>
+        /// <param name="env">The environment to use.</param>
+        /// <param name="lookupMap">An instance of <see cref="IDataView"/> that contains the key and value columns.</param>
+        /// <param name="keyColumn">Name of the key column in <paramref name="lookupMap"/>.</param>
+        /// <param name="valueColumn">Name of the value column in <paramref name="lookupMap"/>.</param>
+        /// <param name="columns">The list of columns to apply.</param>
```

Comment:
>The list of columns to apply [](start = 34, length = 28)

The list of names of the input columns to apply the transformation, and the name of the resulting column.